lib/uknetdev/netbuf.c

```diff
@@ -40,6 +41,42 @@
 #define NETBUF_ADDR_ALIGN_DOWN(x) ALIGN_DOWN((__uptr) (x), \
 					     NETBUF_ADDR_ALIGNMENT)
 
+/**
+ * This function creates a scatter gather list from the netbuf.
+ *
+ * @param sg
+ *	A reference to the scatter gather list.
+ * @param netbuf
+ *	A reference to the netbuf
+ * @return
+ *	0, on successful creation of the scatter gather list
+ *	-EINVAL, Invalid sg list.
+ */
+int uk_netbuf_sglist_append(struct uk_sglist *sg, struct uk_netbuf *netbuf)
```

Comment:
I guess this declaration needs here now to be conditional to `CONFIG_UKSGLIST`

Reply:
There is no sglist dependency by `lib/uknetdev` and I also think we should not introduce it if not neccessary.

Reply:
I believe we do though.
```
drivers/virtio/net/virtio_net.c:34:#include <uk/sglist.h>
drivers/virtio/net/virtio_net.c:98:	struct uk_sglist sg;
drivers/virtio/net/virtio_net.c:99:	struct uk_sglist_seg sgsegs[NET_MAX_FRAGMENTS];
drivers/virtio/net/virtio_net.c:124:	struct uk_sglist sg;
drivers/virtio/net/virtio_net.c:125:	struct uk_sglist_seg sgsegs[NET_MAX_FRAGMENTS];
drivers/virtio/net/virtio_net.c:403:	 *       to `uk_sglist_append_netbuf()`. However, a netbuf
drivers/virtio/net/virtio_net.c:420:	 * Prepare the sglist and enqueue the buffer to the virtio-ring.
drivers/virtio/net/virtio_net.c:422:	uk_sglist_reset(&queue->sg);
drivers/virtio/net/virtio_net.c:432:	rc = uk_sglist_append(&queue->sg, vhdr, virtio_net_hdr_size(vndev));
drivers/virtio/net/virtio_net.c:437:	rc = uk_sglist_append(&queue->sg, buf_start, buf_len);
drivers/virtio/net/virtio_net.c:443:		rc = uk_sglist_append_netbuf(&queue->sg, pkt->next);
drivers/virtio/net/virtio_net.c:451:		total_len = uk_sglist_length(&queue->sg);
drivers/virtio/net/virtio_net.c:504:	struct uk_sglist *sg;
drivers/virtio/net/virtio_net.c:528:	uk_sglist_reset(sg);
drivers/virtio/net/virtio_net.c:530:	/* Appending the header buffer to the sglist */
drivers/virtio/net/virtio_net.c:531:	uk_sglist_append(sg, rxhdr, virtio_net_hdr_size(vndev));
drivers/virtio/net/virtio_net.c:533:	/* Appending the data buffer to the sglist */
drivers/virtio/net/virtio_net.c:534:	uk_sglist_append(sg, buf_start, buf_len);
drivers/virtio/net/virtio_net.c:1087:		uk_sglist_init(&vndev->rxqs[i].sg,
drivers/virtio/net/virtio_net.c:1098:		uk_sglist_init(&vndev->txqs[i].sg,
```

Should I add the dependency in the Config.uk?

Reply:
Hmm ok but `virtio-net` itself depends on `sglist`. So should I move it to `drivers/virtio/net/`?

Reply:
no, you can keep it here, this is fine. Just guard it with
```c
#if LIBUKSGLIST
 /* ... */
#endif /* LIBUKSGLIST */
```
As soon as a driver enables `uksglist`, this driver has then access to this special netbuf function.